controllers/message-utils.js

```diff
@@ -51,45 +128,67 @@ module.exports = {
       callback(null, msgs);
     });
   },
-  updateMessage: function(id, body, callback) {
-    module.exports.getMessage(id, function(err, msg) {
+  /*
+   * See: updateMessageTaskStates
+   */
+  updateMessageTaskState: function(taskStateChange, callback) {
+    module.exports.updateMessageTaskStates([taskStateChange], callback);
+  },
+  /*
+   * taskStateChanges: a collection of:
+   * {
+   *  messageId, state, details
+   * }
+   */
+  updateMessageTaskStates: function(taskStateChanges, callback, retriesLeft=3) {
+    getTaskMessages({ keys: taskStateChanges.map(change => change.messageId)}, (err, taskMessageResults) => {
       if (err) {
         return callback(err);
       }
-      // update requires message_id parameter in JSON body
-      body.message_id = id;
-      updateCouchDB(msg._record_id, body, function(err, data) {
-        if (err) {
-          // Output more informative error message.
-          const isMessageUseless = (message) => !message || message === 'Unspecified error';
-          if (err.message &&
-            isMessageUseless(err.message) &&
-            err.payload &&
-            err.payload.error &&
-            !isMessageUseless(err.payload.error)) {
-            err.message = err.payload.error;
-          }
+
+      const idsToFetch = _.uniq(_.pluck(taskMessageResults.rows, 'id'));
+
+      db.medic.fetch({keys: idsToFetch}, (err, docResults) => {
+        const docs = _.pluck(docResults.rows, 'doc');
+
+        let stateChangesByDocId;
+        try {
+          stateChangesByDocId = applyTaskStateChangesToDocs(taskStateChanges, docs);
+        } catch (err) {
           return callback(err);
         }
-        callback(null, {
-          id: id,
-          success: data.payload.success
+
+        db.medic.bulk({docs: docs}, (err, results) => {
+          if (err) {
+            return callback(err);
+          }
+
+          const failures = results.filter((result) => !result.ok);
```

Comment:
Do we have a preferred style for including/excluding brackets for single-arg anonymous functions?  i.e.

    (result) => !result.ok

vs

    result => !result.ok

?